packages/extensions/venia-sample-eventing/main.js

```diff
@@ -0,0 +1,16 @@
+import { useEffect } from 'react';
+import { useEventingContext } from '@magento/peregrine/lib/context/eventing';
+
+export default original => props => {
+    const [observable, { dispatch }] = useEventingContext();
+
+    useEffect(() => {
+        observable.subscribe(event => {
+            console.log('Logging event:', event);
+        });
+
+        dispatch('hello world');
```

Comment:
We'll be doing most of the dispatching in `peregrine`, but it's possible to do it from an extension.